modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/config/Config.scala

```diff
@@ -166,6 +166,16 @@ object Config {
           }
       }
     }
+
+    final case class Azure(blobStorageEndpoint: URI, messageQueue: Azure.Kafka) extends Cloud
+
+    object Azure {
+      final case class Kafka(
+        topicName: String,
+        bootstrapServers: String,
+        consumerConf: Map[String, String]
```

Comment:
Is there any important configuration parameter which you think we should pass in the consumer conf?  Or is this here just in case we discover we need it in future?

Reply:
To connect to event hubs we have to provide certain properties to kafka consumer/producer, sth lik: 

```
"security.protocol": "SASL_SSL",
"sasl.mechanism": "PLAIN",
"sasl.jaas.config": "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"$ConnectionString\" password=...."
```

Reply:
Ah I see.  I like that design a lot, to pass through the standard configuration options.  It gives maximum configurability to the user, and the config options are well documented elsewhere.

I think we should put a few sensible defaults into our `application.conf` file, so that the default behaviour matches snowplow pipeline requirements.

- `group.id` could be set to something like `snowplow-transformer`.  If left empty, then all transformer instances would see all events in the topic, and this is definitely not desired behaviour.
- `allow.auto.create.topics` should be false.  This is to be consistent with how we manage all other infrastructure at Snowplow.  We trust the terraform to create the cloud resources.
- `auto.offset.reset` should be probably `latest`, but definitely not `none`.  Note the kafka default is latest, but fs2-kafka overrides the default and sets it to none by default.  This would be a problem for new pipelines which are not receiving events yet.

For producer settings the defaults look reasonable, so we probably don't need anything in `application.conf` for producers.